src/adapter/src/catalog.rs

```diff
@@ -2849,6 +2850,114 @@ impl Catalog {
                         )?;
                     }
                 }
+                Op::RenameSchema {
+                    database_spec,
+                    schema_spec,
+                    new_name,
+                } => {
+                    let ResolvedDatabaseSpecifier::Id(database_id) = &database_spec else {
+                        let err = anyhow::anyhow!("Cannot rename schemas in the ambient database");
+                        return Err(AdapterError::Unstructured(err));
+                    };
+                    let SchemaSpecifier::Id(schema_id) = &schema_spec else {
+                        let err = anyhow::anyhow!("Cannot rename temporary schemas");
+                        return Err(AdapterError::Unstructured(err));
+                    };
+
+                    let database = state.get_database(database_id);
+                    let schema = database.schemas_by_id.get(schema_id).ok_or_else(|| {
+                        let err = anyhow::anyhow!("unknown schema");
+                        AdapterError::Unstructured(err)
```

Comment:
This seems like it's unreachable. If you were previously able to resolve the schema in the database then it must still exist. Still, I think we should use a structured error here. `mz_sql::catalog::CatalogError::UnknownSchema` seems appropriate.

Reply:
Refactored it away